### ISSUE TITLE
Oban 2.2.0 doesn't start all it's children

### ISSUE THREAD

**Issue by moogle19** [sorentwo/oban#356]
### Environment

- Oban Version: `2.2.0`
- PostgreSQL Version: `13.0`
- Elixir & Erlang/OTP Versions (`elixir --version`): `Elixir 1.10.4 / Erlang 22`

### Current Behavior
Children (e.g. Notifier) don't start on application start.
I am not sure, if v2.2 changed the configuration for this somehow (the Changelog didn't mention anything about that).

This is the overview of Oban 2.2.0 processes in the Live Dashboard:
<img width="1421" alt="Screen Shot 2020-10-13 at 3 47 00 PM" src="https://user-images.githubusercontent.com/792046/95870415-be950b80-0d6c-11eb-9149-7e0eff9613ee.png">

In Oban 2.1.0 it was like this:
<img width="1288" alt="Screen Shot 2020-10-13 at 3 51 46 PM" src="https://user-images.githubusercontent.com/792046/95870514-dcfb0700-0d6c-11eb-8fe3-a7c0ddf14a0e.png">

Listening for notifications in v2.2.0 fails with:
```
** (exit) exited in: GenServer.call(Oban.Notifier, {:listen, [:insert, :signal, :gossip]}, 5000)
** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
```

My config didn't change between the two version:
```
config :my_application, Oban,
  repo: MyApplication.Repo,
  plugins: [{Oban.Plugins.Pruner, max_age: 2_592_000}], # Set Pruner with max_age: 1 month
  queues: [default: 4, events: 4, media: 4]
```
I start Oban with:
```
def start(_type, _args) do
    children = [
      MyApplication.Repo,
      ... other children ...
      {Oban, oban_config()}
    ]

    opts = [strategy: :one_for_one, name: MyApplication.Supervisor]
    Supervisor.start_link(children, opts)
  end

  defp oban_config do
    Application.get_env(:my_application, Oban)
  end
```

### Expected Behavior

All children should be started (like in v2.1.0)

**Comment by sorentwo** [comment]
The processes shown in your first screen shot are from the Oban application, which has the global registry. It doesn't look like your Oban supervisor instance has started at all.

Did you change anything else or are you getting a crash report?

**Comment by moogle19** [comment]
Nope, just updated from 2.1 to 2.2. 
On first glance I didn’t see an error message. I will check tomorrow when I’m at work.

**Comment by moogle19** [comment]
My findings so far are:
The `listen` function in `Oban.Notifier` will not work, because the `name` of the `GenServer` is not `Oban.Notifier` anymore but `{:via, Registry, {Oban.Registry, {Oban, Oban.Notifier}}}`

Splitting the `listen` function to handle `pid`s and `atom`s seperatly didn't work at first. I thought calling  `Oban.Registry.whereis`/`Oban.Registry.via` with `Oban.Notifier` would return the correct name, but it didn't (it returned `{:via, Registry, {Oban.Registry, Oban.Notifier}}`)

My current solution is splitting the `listen` function as follows:

```elixir
# Handle Pids
def listen(pid, channels) when is_pid(pid) and is_list(channels) do
    :ok = validate_channels!(channels)

    GenServer.call(pid, {:listen, channels})
  end

# Handle atoms
def listen(server \\ __MODULE__, channels) when is_atom(server) and is_list(channels) do
  :ok = validate_channels!(channels)

  # Fetch Notifier PID from Registry
  pid = Oban.Registry.whereis(Oban, Oban.Notifier)

  GenServer.call(pid, {:listen, channels})
end
```

Not sure if this is how thing should be handled.
If so I could make a Pull Request with my changes.

**Comment by moogle19** [comment]
Regarding the missing processes in the Live Dashboard I found, that Observer show the right name, but the Live Dashboard doesn't.

__Observer:__
<img width="813" alt="Screen Shot 2020-10-14 at 10 41 57 AM" src="https://user-images.githubusercontent.com/792046/95965375-277e9100-0e0a-11eb-9766-e08206c586c4.png">
__Live Dashboard:__
<img width="1277" alt="Screen Shot 2020-10-14 at 10 41 46 AM" src="https://user-images.githubusercontent.com/792046/95965381-28afbe00-0e0a-11eb-9b1c-898bf94e4769.png">

**Comment by sorentwo** [comment]
Thanks for the investigation! Now I see the issue—the process is there, and internally Oban is calling it correctly, but calling `listen/2` from the outside no longer works with a module name. I think the best course of action is to add a new clause to `listen/2`, as you started doing (and be sure to update the documentation accordingly).

A PR is very welcome.

**Comment by sorentwo** [comment]
Closed in favor of the PR, #358
